presto-tpch/src/main/java/com/facebook/presto/tpch/StatisticsEstimator.java

```diff
@@ -0,0 +1,258 @@
+        else {
+            Number left = checkType(leftValue, Number.class);
+            Number right = checkType(rightValue, Number.class);
+            return areClose(left.doubleValue(), right.doubleValue());
+        }
+    }
+
+    private boolean areClose(double smallCount, double bigCount)
+    {
+        double difference = abs(bigCount - smallCount);
+        return difference <= abs(smallCount) * 0.1 || difference <= 0;
+    }
+
+    private TableStatisticsData readStatistics(String tableName, Predicate<Map<TpchColumn<?>, Object>> constraint, String scaleFactor)
+    {
+        //TODO try to pull this up, make counts checkPresent() for nonscaling tables
+        if (tableName.equals(ORDERS.getTableName())) {
+            if (Stream.of("F", "O", "P").allMatch(value -> constraint.test(ImmutableMap.of(ORDER_STATUS, value)))) {
+                return parseStatistics(tableName, scaleFactor);
+            }
+
+            TableStatisticsData result = emptyStatistics(ORDERS);
+            result = addOrderPartition(result, constraint, scaleFactor, "F");
+            result = addOrderPartition(result, constraint, scaleFactor, "O");
+            result = addOrderPartition(result, constraint, scaleFactor, "P");
+
+            return result;
+        }
+        else {
+            return parseStatistics(tableName, scaleFactor);
+        }
+    }
+
+    private TableStatisticsData addOrderPartition(TableStatisticsData result, Predicate<Map<TpchColumn<?>, Object>> constraint, String scaleFactor, String partitionValue)
+    {
+        if (constraint.test(ImmutableMap.of(ORDER_STATUS, partitionValue))) {
+            result = addPartitionStats(result, parseStatistics("orders.o_orderstatus." + partitionValue, scaleFactor), ORDER_STATUS);
+        }
+        return result;
+    }
+
+    private TableStatisticsData addPartitionStats(TableStatisticsData left, TableStatisticsData right, TpchColumn<?> partitionColumn)
+    {
+        return new TableStatisticsData(
+                left.getRowCount() + right.getRowCount(),
+                addPartitionStats(left.getColumns(), right.getColumns(), partitionColumn)
+        );
+    }
+
+    private Map<String, ColumnStatisticsData> addPartitionStats(Map<String, ColumnStatisticsData> leftColumns, Map<String, ColumnStatisticsData> rightColumns, TpchColumn<?> partitionColumn)
+    {
+        return leftColumns.entrySet().stream().collect(toImmutableMap(
+                Map.Entry::getKey,
+                entry -> {
+                    String columnName = entry.getKey();
+                    ColumnStatisticsData leftStats = entry.getValue();
+                    ColumnStatisticsData rightStats = rightColumns.get(columnName);
+                    return new ColumnStatisticsData(
+                            combine(leftStats.getUniqueValuesCount(), rightStats.getUniqueValuesCount(), (a, b) -> a + b)
+                                    .filter(v -> columnName.equals(partitionColumn.getColumnName())),
```

Comment:
Did this:
```java
    private Optional<Long> combineUniqueValuesCount(TpchColumn<?> partitionColumn, String columnName, ColumnStatisticsData leftStats, ColumnStatisticsData rightStats)
    {
        //unique values count can't be added between different partitions
        //for columns other than the partition column (because almost certainly there are duplicates)
        return combine(leftStats.getUniqueValuesCount(), rightStats.getUniqueValuesCount(), (a, b) -> a + b)
                .filter(v -> columnName.equals(partitionColumn.getColumnName()));
    }
```